CameraPanel/CameraAndroid.uno

```diff
@@ -42,24 +42,42 @@ public extern(Android) class Camera
 		cameraAndroid.setOnFrameAvailableListener(new android.graphics.SurfaceTexture.OnFrameAvailableListener() {
 			public void onFrameAvailable(android.graphics.SurfaceTexture surfaceTexture) {
 				if(cameraAndroid.HasSurfaceTexture())
-				{
-					@{Camera:Of(_this).MakeContextCurrent():Call()};
-					surfaceTexture.updateTexImage();
-					@{Camera:Of(_this).OnFrameAvailable():Call()};
-				}
+					@{Camera:Of(_this).DispatchFrameAvailable(Java.Object):Call(surfaceTexture)};
 			}
 		});
 
 		android.app.Activity activity = @(Activity.Package).@(Activity.Name).GetRootActivity();
 		cameraAndroid.start(activity.getWindowManager().getDefaultDisplay().getRotation(), textureHandle, @{Camera:Of(_this).IntFacing:Get()});
 	@}
 
-	void MakeContextCurrent()
+	class FrameAvailableClosure
 	{
-		if defined(Android)
-			extern "GLHelper::MakeWorkerThreadContextCurrent()";
+		Camera _camera;
+		Java.Object _tex;
+
+		public FrameAvailableClosure(Camera camera, Java.Object tex)
+		{
+			_camera = camera;
+			_tex = tex;
+		}
+
+		public void FrameAvailable()
+		{
+			UpdateTexImage(_tex);
```

Comment:
Will complete my answer once we've been able to re-test on the Samsung Trend Plus, where the preview did not work since the app launch. Should be done with the next 24 hours